server/metrics.go

```diff
@@ -13,10 +13,36 @@
 package server
 
 import (
+	"github.com/prometheus/client_golang/prometheus"
 	metrics "github.com/slok/go-http-metrics/metrics/prometheus"
 	"github.com/slok/go-http-metrics/middleware"
 )
 
 var prometheusMiddleware = middleware.New(middleware.Config{
 	Recorder: metrics.NewRecorder(metrics.Config{}),
 })
+
+var totalRegisteredPolicies = prometheus.NewCounter(
+	prometheus.CounterOpts{
+		Namespace: "krail",
+		Name:      "total_registered_policies",
+		Help:      "Total Policies Registered",
+	},
+)
+
+var totalLoadedPlugins = prometheus.NewCounter(
+	prometheus.CounterOpts{
+		Namespace: "krail",
+		Name:      "total_loaded_plugins",
+		Help:      "Total Plugins Loaded",
+	},
+)
+
+var policyViolations = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Namespace: "krail",
+		Name:      "policy_violations",
+		Help:      "Count of Violations",
+	},
+	[]string{"kind", "resource", "namespace", "policy", "user", "enforced", "exempt", "report_only", "global_report_only"},
```

Comment:
This amount of labels can be dangerous if there are a lot of violations, because each unique combination of labels creates a timeseries in prometheus. Many timeseries can cause performance issues, and can also be expensive with services like Datadog.

Reply:
Ahhh yeah that does make sense. Maybe I can narrow it down to just the policy and whether it was enforced or not.

Reply:
Good point Dustin, could we make it configurable (in the yaml config file) with the default being a narrowed down list?

Reply:
I don't see why we can't, we'll just want to let people know with additional metrics comes potential for slow downs and expensive metric storage. Maybe make it enabled/disabled per policy in the configuration file?

If we did keep the metric list basic, logs would be the supporting artifacts to figure out different situations, for example when there is a jump in policy violations. Use this to alert on, then dig into the logs to find the answer.

Reply:
After going through different iterations of adding more complex code for metrics, seems like reducing the labels to basic needs is all we need `resource`, `namespace`, `policy`, `enforced` + adding a top level configuration `global_metrics_enabled: True`

Two situations metrics helps with:
- If k-rail is new to the cluster then the goal would be to work with the person or team on policy violations - based on violations from metrics.
- If k-rail is not new to the cluster then the goal would be to alert on policy violations.